### PR TITLE
Ensured ``changed`` returns ``False``.

### LINKED ISSUE #1040: ldap_search sets changed to true, and shouldn't
##### SUMMARY
ldap_search always sets changed to true.  But searching never changes anything so changed should always be false.

This is true in HEAD as of this writing, although I did not report that in the ansible version below.  I read the code.
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ldap_search

##### ANSIBLE VERSION

```paste below
ansible 2.10.1
  config file = /home/ubuntu/ansible/ansible.cfg
  configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ubuntu/ansible/venv/lib/python3.6/site-packages/ansible
  executable location = venv/bin/ansible
  python version = 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
```

##### CONFIGURATION

```paste below
COLLECTIONS_PATHS(/home/ubuntu/ansible/ansible.cfg) = ['/home/ubuntu/ansible/col
DEFAULT_BECOME(/home/ubuntu/ansible/ansible.cfg) = True
INTERPRETER_PYTHON(/home/ubuntu/ansible/ansible.cfg) = /usr/bin/python3
```

##### OS / ENVIRONMENT

Linux READACTED 5.4.0-1025-aws #25~18.04.1-Ubuntu SMP Fri Sep 11 12:03:04 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

##### STEPS TO REPRODUCE

```yaml
- hosts: localhost
  gather_facts:	false
  tasks:
  - community.general.ldap_search:
      attrs: dn
      dn: "cn=schema,cn=config"
      scope: onelevel
      filter: "(cn=*}nis)"

```

##### EXPECTED RESULTS

The task should set changed to false.  Also when: filter: "(cn=nonexistant)"

##### ACTUAL RESULTS

The task sets changed to true.

```paste below
PLAY [redacted] ******************************************************

TASK [community.general.ldap_search] *******************************************
changed: [redacted]

PLAY RECAP *********************************************************************
redacted    : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

```

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Made the change.

Also added a small improvement on the ``_load_scope()`` method.

Fixes #1040 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ldap_search

### CODE REVIEW

File: changelogs/fragments/1040-ldap_search-changed-must-be-false.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
```
Comment (felixfontein) [inline changelogs/fragments/1040-ldap_search-changed-must-be-false.yaml:1]:
```suggestion
breaking_change:
```

File: changelogs/fragments/1040-ldap_search-changed-must-be-false.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+  - ldap_search - replaced the output value ``changed`` with ``False`` (https://github.com/ansible-collections/community.general/issues/1040).
```
Comment (felixfontein) [inline changelogs/fragments/1040-ldap_search-changed-must-be-false.yaml:2]:
```suggestion
  - ldap_search - the module no longer reports a change (https://github.com/ansible-collections/community.general/issues/1040).
```

Comment (russoz) [reply changelogs/fragments/1040-ldap_search-changed-must-be-false.yaml:2]: Well, maybe "the module no longer lies about having a change to report" :-)

Comment (felixfontein) [reply changelogs/fragments/1040-ldap_search-changed-must-be-false.yaml:2]: `no longer incorrectly reports a change`?

**Comment by felixfontein** [comment]
@russoz thanks for fixing this!

**Comment by russoz** [comment]
@felixfontein thanks for reviewing :grin:

**Pull request merged**